adb/adb_debug.py

```diff
@@ -36,34 +37,144 @@
     rsa_signer = None
 
 
-gflags.ADOPT_module_key_flags(common_cli)
+def Devices(args):
+  """Lists the available devices.
+
+  Mimics 'adb devices' output:
+    List of devices attached
+    015DB7591102001A        device        1,2
+  """
+  for d in adb_commands.AdbCommands.Devices():
+    if args.output_port_path:
+      print('%s\tdevice\t%s' % (
+            d.serial_number, ','.join(str(p) for p in d.port_path)))
+    else:
+      print('%s\tdevice' % d.serial_number)
+  return 0
+
+
+def List(self, device_path):
+  """Prints a directory listing.
+
+  Args:
+    device_path: Directory to list.
+  """
+  files = adb_commands.AdbCommands.List(self, device_path)
+  files.sort(key=lambda x: x.filename)
+  maxname = max(len(f.filename) for f in files)
+  maxsize = max(len(str(f.size)) for f in files)
+  for f in files:
+    mode = (
+      ('d' if stat.S_ISDIR(f.mode) else '-') +
+      ('r' if f.mode & stat.S_IRUSR else '-') +
+      ('w' if f.mode & stat.S_IWUSR else '-') +
+      ('x' if f.mode & stat.S_IXUSR else '-') +
+      ('r' if f.mode & stat.S_IRGRP else '-') +
+      ('w' if f.mode & stat.S_IWGRP else '-') +
+      ('x' if f.mode & stat.S_IXGRP else '-') +
+      ('r' if f.mode & stat.S_IROTH else '-') +
+      ('w' if f.mode & stat.S_IWOTH else '-') +
+      ('x' if f.mode & stat.S_IXOTH else '-'))
+    t = time.gmtime(f.mtime)
+    yield '%s %*d %04d-%02d-%02d %02d:%02d:%02d %-*s\n' % (
+        mode, maxsize, f.size,
+        t.tm_year, t.tm_mon, t.tm_mday, t.tm_hour, t.tm_min, t.tm_sec,
+        maxname, f.filename)
+
+
+@functools.wraps(adb_commands.AdbCommands.Logcat)
+def Logcat(self, *options):
+  return adb_commands.AdbCommands.StreamingShell(
+      self, 'logcat ' + ' '.join(options))
+
+
+def Shell(self, *command):
+  """Runs a command on the device and prints the stdout.
+
+  Args:
+    command: Command to run on the target.
+  """
+  return adb_commands.AdbCommands.StreamingShell(self, ' '.join(command))
+
+
+def main():
+  common = common_cli.GetCommonArguments()
+  common.add_argument(
+      '--rsa_key_path', action='append', default=[],
+      metavar='~/.android/adbkey',
+      help='RSA key(s) to use, use multiple times to load mulitple keys')
+  common.add_argument(
+    '--auth_timeout_s', default=60., metavar='60', type=int,
```

Comment:
I like that the output is:
```
  --auth_timeout_s 60   Seconds to wait for the dialog to be accepted when
                        using authenticated ADB.
````

It's clear (to me) that it is the default value.